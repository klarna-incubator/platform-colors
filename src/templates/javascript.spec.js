const generate = require('./javascript');
const parseColorManifest = require('../parse-color-manifest');
const colors = parseColorManifest(require('./fixtures/light-colors.json'));

const getFileContents = (files, filename) => {
  const file = files.find((file) => file[0] === filename);
  if (!file) {
    throw new Error(`File "${filename}" not present in output`);
  }
  return file[1];
};

const expectFileSnapshot = (config, filename) => {
  const files = generate(colors, config);
  const contents = getFileContents(files, filename);
  expect(contents).toMatchSnapshot();
};

describe('javascript template', () => {
  it('outputs ios implementation if present in config', () => {
    expectFileSnapshot({ ios: {} }, 'index.ios.js');
  });

  it('outputs android implementation if present in config', () => {
    expectFileSnapshot({ android: {} }, 'index.android.js');
  });

  it('outputs type definitions if enabled in config', () => {
    expectFileSnapshot(
      { android: {}, javascript: { typescript: true } },
      'index.d.ts'
    );
  });

  it('outputs web implementation if present in config', () => {
    expectFileSnapshot({ css: {} }, 'index.js');
  });

  it('outputs typescript for web if enabled in config', () => {
    expectFileSnapshot(
      { css: {}, javascript: { typescript: true } },
      'index.ts'
    );
  });
});
