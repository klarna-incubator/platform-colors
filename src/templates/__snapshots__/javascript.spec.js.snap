// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`javascript template outputs android implementation if present in config 1`] = `
"import { PlatformColor } from 'react-native';

export const stringColor = PlatformColor('@color/rnpc_string_color');
export const alphaColor = PlatformColor('@color/rnpc_alpha_color');
export const namedColor = PlatformColor('@color/rnpc_named_color');"
`;

exports[`javascript template outputs ios implementation if present in config 1`] = `
"import { PlatformColor } from 'react-native';

export const stringColor = PlatformColor('RnpcStringColor');
export const alphaColor = PlatformColor('RnpcAlphaColor');
export const namedColor = PlatformColor('RnpcNamedColor');"
`;

exports[`javascript template outputs type definitions if enabled in config 1`] = `
"import { OpaqueColorValue } from 'react-native';

export const stringColor: OpaqueColorValue;
export const alphaColor: OpaqueColorValue;
export const namedColor: OpaqueColorValue;"
`;

exports[`javascript template outputs typescript for web if enabled in config 1`] = `
"export const stringColor = 'var(--rnpc-string-color)';
export const alphaColor = 'var(--rnpc-alpha-color)';
export const namedColor = 'var(--rnpc-named-color)';"
`;

exports[`javascript template outputs web implementation if present in config 1`] = `
"export const stringColor = 'var(--rnpc-string-color)';
export const alphaColor = 'var(--rnpc-alpha-color)';
export const namedColor = 'var(--rnpc-named-color)';"
`;
